### PR TITLE
[BPK-2427] Add BPKPrimaryGradientView

### PR DESCRIPTION
**Pull request by k0nserv**
<details>
<summary>London</summary>

Simulator Screen Shot - iPhone X - 2019-04-29 at 13 20 54

</details>

<details>
<summary>Doha</summary>

Simulator Screen Shot - iPhone X - 2019-04-29 at 13 20 49

</details>

<details>
<summary>Hong Kong</summary>

Simulator Screen Shot - iPhone X - 2019-04-29 at 13 20 52

</details>

+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-ios/blob/master/CONTRIBUTING.md)

### CODE REVIEW

File: Backpack/Gradient/Classes/BPKPrimaryGradientView.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+NS_ASSUME_NONNULL_BEGIN
+@class BPKGradient;
+
+/**
+ * This class is a `UIView` subclass that draws the primary gradient.
+ * It's backed by a layer that does the actual drawing, but using this view directly
+ * is preferred.
+ */
+IB_DESIGNABLE NS_SWIFT_NAME(PrimaryGradientView) @interface BPKPrimaryGradientView : UIView
+/**
```
Comment (georgegillams) [inline Backpack/Gradient/Classes/BPKPrimaryGradientView.h]: Space above this line?

File: Backpack/Gradient/Classes/BPKPrimaryGradientView.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * The gradient currently being drawn.
+ * @warning Do not set this value directly. It's intended to be used via theming.
+ */
+@property(nonatomic, strong) BPKGradient *gradient UI_APPEARANCE_SELECTOR;
+@end
```
Comment (georgegillams) [inline Backpack/Gradient/Classes/BPKPrimaryGradientView.h]: space needed here?

File: CHANGELOG.md [hunk 3]
```diff
@@ -7,10 +7,10 @@
 **Added:**
 
 - Backpack/Theme:
-  - Added `buttonCornerRadius` properties to Backpack theme definitions.
+  - Added `buttonCornerRadius` property to Backpack theme definitions.
```
Comment (georgegillams) [inline CHANGELOG.md]: Good catch!

**Comment by k0nserv** [comment]
> You got snapshot tests working with theming? Awesome!

No I just set the themeable property manually. I think we coud create something that can set all the themeable properties for an instance of a view or an appearance proxy however. This way we could test theming fairly well

File: Example/Backpack/View Controllers/GradientViewController.swift [hunk 4]
```diff
@@ -20,20 +20,17 @@ import UIKit
...
 
 enum GradientType {
-    case primary
     case baselinesSrim
 }
 
 class GradientViewController: UIViewController {
     @IBOutlet weak var gradientView: Backpack.GradientView!
-    var gradientType: GradientType = .primary
+    var gradientType: GradientType = .baselinesSrim
```
Comment (shaundon) [inline Example/Backpack/View Controllers/GradientViewController.swift]: Is this a typo? `Srim` seems weird

Comment (k0nserv) [reply Example/Backpack/View Controllers/GradientViewController.swift]: That's exactly what it looks like. Fixed now

**Pull request merged**